torch/csrc/utils/python_numbers.h

```diff
@@ -100,30 +106,28 @@ inline bool THPUtils_unpackBool(PyObject* obj) {
 }
 
 inline bool THPUtils_checkDouble(PyObject* obj) {
-  bool is_numpy_scalar;
 #ifdef USE_NUMPY
-  is_numpy_scalar = torch::utils::is_numpy_scalar(obj);
-#else
-  is_numpy_scalar = false;
+  if (torch::utils::is_numpy_scalar(obj) && ! PyBool_Check(obj)) {
+    return true;
+  }
 #endif
 #if PY_MAJOR_VERSION == 2
-  return PyFloat_Check(obj) || PyLong_Check(obj) || PyInt_Check(obj) || is_numpy_scalar;
+  return PyFloat_Check(obj) || PyLong_Check(obj) || PyInt_Check(obj);
 #else
-  return PyFloat_Check(obj) || PyLong_Check(obj) || is_numpy_scalar;
+  return PyFloat_Check(obj) || PyLong_Check(obj);
 #endif
 }
 
 inline bool THPUtils_checkScalar(PyObject* obj) {
-  bool is_numpy_scalar;
 #ifdef USE_NUMPY
-  is_numpy_scalar = torch::utils::is_numpy_scalar(obj);
-#else
-  is_numpy_scalar = false;
+  if (torch::utils::is_numpy_scalar(obj)) {
```

Comment:
I didn't change the behavior for checkScalar (since it wasn't _directly_ related to the reported bug), but it looks like the existing code would return true for a numpy boolean where it probably shouldn't (the below doesn't seem to include boolean).

I'm not very familiar with how we expect complex types to work in pytorch.

Reply:
I checked that `is_numpy_scalar` returns false for np.bool